src/gifcodec.c

```diff
@@ -198,40 +191,46 @@ DGifSlurpMono(GifFileType * GifFile, SavedImage *TrailingExtensions)
 
 		switch (RecordType) {
 			case IMAGE_DESC_RECORD_TYPE: {
-				/* This call will leak GifFile->Image.ColorMap; there's a fixme in the DGifGetImageDesc code */
 				if (DGifGetImageDesc(GifFile) == GIF_ERROR) {
 					return (GIF_ERROR);
 				}
 
 				sp = &GifFile->SavedImages[GifFile->ImageCount - 1];
+				/* Allocate memory for the image */
+				if (sp->ImageDesc.Width < 0 && sp->ImageDesc.Height < 0 &&
+						sp->ImageDesc.Width > (INT_MAX / sp->ImageDesc.Height)) {
+					return GIF_ERROR;
+				}
+
 				ImageSize = sp->ImageDesc.Width * sp->ImageDesc.Height;
-				if (ImageSize == 0)
+				if (ImageSize > (SIZE_MAX / sizeof(GifPixelType))) {
```

Comment:
...and this fails the test on CI (testgifcodec, `createFile (zeroImageWidth, OutOfMemory);`). Maybe `if (ImageSize == 0 || ImageSize > (SIZE_MAX / sizeof(GifPixelType)))` ?